docs: add tailwind-css instructions (#88)
electron-forge/electron-forge-docs: guides/framework-integration/tailwind.md

---
description: How to create an Electron app with Tailwind and Electron Forge
---

# Tailwind CSS

> Adding Tailwind CSS support to the Webpack template involves adding postcss support to the webpack configuration.

Reviewer: suggestion: link to other projects and capitalize as appropriate.
```suggestion
Adding [Tailwind CSS](https://tailwindcss.com/) support to an Electron Forge project using the [Webpack plugin](../../config/plugins/webpack.md) involves adding [PostCSS](https://postcss.org/) support to the Webpack configuration.
```

{% hint style="info" %}
The following guide has been tested with Tailwind 3, and Webpack 5.
{% endhint %}

### Create the app and install Tailwind CSS

Create the app with the [Webpack template](../../templates/webpack-template.md).

> Install Tailwind CSS using the [`postcss instructions`](https://tailwindcss.com/docs/installation/using-postcss).

Reviewer 2: IMO we might want to port the instructions over here

> Create the app with the [Webpack template](../../templates/webpack-template.md).
> 
> Install Tailwind CSS using the [`postcss instructions`](https://tailwindcss.com/docs/installation/using-postcss).

Reviewer 2: ````suggestion
Starting with the Electron Forge [Webpack Template](../../templates/webpack-template.md), install Tailwind CSS using the [PostCSS instructions](https://tailwindcss.com/docs/installation/using-postcss).

{% tabs %}
{% tab title="Yarn" %}
```bash
yarn add --dev tailwindcss postcss autoprefixer
npx tailwindcss init -p
```
{% endtab %}

{% tab title="npm" %}
```bash
npm install --save-dev tailwindcss postcss autoprefixer
npx tailwindcss init -p
```
{% endtab %}
{% endtabs %}

Running the `npx tailwindcss init -p` script will create new `tailwind.config.js` and `postcss.config.js` configs for you. After this code is initialized, make sure you include all source files in your Tailwind `content` config and the necessary `@tailwind` directives to your stylesheet.

{% tabs %}
{% tab title="tailwind.config.js" %}
```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ["./src/**/*.{html,js}"],
  theme: {
    extend: {},
  },
  plugins: [],
}

```
{% endtab %}

{% tab title="src/index.css" %}
```css
@tailwind base;
@tailwind components;
@tailwind utilities;
```
{% endtab %}
{% endtabs %}

````

### Configure Webpack to use postcss

> Install the postcss-loader component:

Reviewer: Nit(formatting):
```suggestion
Install the `postcss-loader` component:
```

{% tabs %}
{% tab title="Yarn" %}
```bash
yarn add --dev postcss-loader
```
{% endtab %}

{% tab title="NPM" %}
```text
npm install --save-dev postcss-loader
```
{% endtab %}
{% endtabs %}

> Modify the renderer webpack configuration to include the postcss-loader for its css rules in `webpack.renderer.config.js`:

Reviewer: Nit(formatting):
```suggestion
Modify the renderer Webpack configuration to include the `postcss-loader` for its CSS rules in `webpack.renderer.config.js`:
```

{% code title="webpack.renderer.config.js" %}
```javascript
rules.push({
  test: /\.css$/,

>   // This is the line that needs modified - by adding the postcss-loader to the list of css rules

Reviewer: Nit(grammar):
```suggestion
  // Modify the following line by adding the postcss-loader module to the list of loaders for CSS files
```

  use: [{ loader: 'style-loader' }, { loader: 'css-loader' }, { loader: 'postcss-loader'}],
});
```
{% endcode %}

### Integrate Tailwind CSS styling

You should now be able to start writing and using Tailwind CSS styling in your Electron app. To confirm, add the Tailwind CSS classes to the default `src/index.html`:

{% code title="src/index.html" %}
```html
  <body>
    <h1 class="text-3xl font-bold underline text-green-500">💖 Hello World!</h1>
    <p>Welcome to your Electron application.</p>
  </body>
```
{% endcode %}

For more about Tailwind CSS, see [their documentation](https://tailwindcss.com/docs).

> ### Using with React and Typescript

Reviewer: Nit(capitalization):
```suggestion
### Using with React and TypeScript
```

> These instructions also work for react, typescript, and react + typescript projects. Follow these instructions after confirming that react and typescript are configured and rendering properly. 

Reviewer: suggestion: capitalize project names accordingly, fix up grammar, and link to the appropriate guides.
```suggestion
These instructions also work for the [TypeScript + Webpack template](../../templates/typescript-%2B-webpack-template.md)  and the [React](react.md) and [React + TypeScript](react-with-typescript.md) guides. Follow these instructions after confirming that TypeScript and/or React are configured and rendering properly. 
```
